drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -1157,6 +1161,22 @@ void nrf_802154_serialization_error(const nrf_802154_ser_err_data_t *err)
 }
 #endif
 
+#if defined(CONFIG_IEEE802154_NRF5_MULTIPLE_CCA)
+void z_ieee802154_nrf5_extra_cca_attempts_set(const struct device *dev, uint8_t value)
+{
+	struct nrf5_802154_data *nrf5_radio = NRF5_802154_DATA(dev);
+
+	nrf5_radio->extra_cca_attempts = value;
```

Comment:
IMO this is a potential race condition (especially as this is uint8_t which might provoke undefined behavior of neighbouring fields on 32bit archs while concurrently accessed by multiple threads if I'm not mistaken). Maybe you need to introduce locking or do this with atomic members?

If this is to be set on a per-packet basis as in your initial PR then I also think that such an indirection may make it difficult to read/maintain that code. A solution inside net_pkt (as a driver-specific extension) would be probably less error prone? It also implicitly solves the concurrency issue as inside net_pkt this field would be immutable.

Reply:
On nrf5 family we have cortex-m4 and cortex-m33, the store is perfromed with one `strb` instruction, this does not affect neighboring moemory calls.
However as pointed in code proposed in comment https://github.com/zephyrproject-rtos/zephyr/pull/60390#issuecomment-1635701165 the transmission is not performed in a single call. 
It is caller responsibility to ensure there is no race condition on access to radio api.

Reply:
> It is caller responsibility to ensure there is no race condition on access to radio api.

As long as we don't make this a general requirement for all drivers implementing the radio API but a requirement concerning the nRF5 driver only, that is your choice of course.